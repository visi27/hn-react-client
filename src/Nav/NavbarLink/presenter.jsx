import React from 'react';
import { PropTypes } from 'prop-types';

const NavbarLink = ({
  currentLocation, href, disabled, children, history,
}) => {
  let className = 'nav-item';
  className += href === currentLocation && !disabled ? ' active' : '';
  const disabledClass = disabled ? ' disabled' : '';

  return (
    <li className={className}>
      <a
        className={`nav-link ${disabledClass}`}
        href={href}
        onClick={(e) => {
          e.preventDefault();
          history.push(href);
        }}
      >
        {children}
      </a>
    </li>
  );
};

NavbarLink.propTypes = {
  currentLocation: PropTypes.string,
  href: PropTypes.string,
  disabled: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  history: PropTypes.shape({}).isRequired,
};

NavbarLink.defaultProps = {
  currentLocation: '/',
  href: '#',
  disabled: false,
  children: 'Menu',
};

export default NavbarLink;
