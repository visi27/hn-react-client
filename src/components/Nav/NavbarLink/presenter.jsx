import React from 'react';
import { PropTypes } from 'prop-types';

const NavbarLink = ({
  itemKey, activeKey, href, disabled, children, onClick,
}) => {
  let className = 'nav-item';
  className += itemKey === activeKey && !disabled ? ' active' : '';
  const disabledClass = disabled ? ' disabled' : '';

  return (
    <li className={className}>
      <a className={`nav-link ${disabledClass}`} href={href} onClick={() => onClick(itemKey)}>
        {children}
      </a>
    </li>
  );
};

NavbarLink.propTypes = {
  itemKey: PropTypes.string,
  activeKey: PropTypes.string,
  href: PropTypes.string,
  disabled: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
  onClick: PropTypes.func,
};

NavbarLink.defaultProps = {
  itemKey: '',
  activeKey: '',
  href: '#',
  disabled: false,
  children: 'Menu',
  onClick: () => {},
};

export default NavbarLink;
