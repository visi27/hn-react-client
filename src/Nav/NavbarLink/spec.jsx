import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import NavbarLink from './presenter';

configure({ adapter: new Adapter() });

describe('Nav', () => {
  const props = {
    currentLocation: '/',
    href: '/',
    disabled: false,
    children: 'testLink',
    history: {},
  };

  const propsActiveLink = { ...props, activeKey: 'testLink' };
  const propsDisabledLink = { ...props, disabled: true };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(<NavbarLink {...props} />, div);
  });

  test('has valid snapshot', () => {
    const component = renderer.create(<NavbarLink {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows 1 menu item', () => {
    const element = mount(<NavbarLink {...props} />);

    expect(element.find('.nav-item').length).toBe(1);
    expect(element.find('.nav-link').length).toBe(1);

    expect(element.find('.nav-item active').length).toBe(0);
  });

  it('shows 1 active menu item', () => {
    const element = mount(<NavbarLink {...propsActiveLink} />);
    expect(element.find('.nav-item .active').length).toBe(1);
  });

  it('shows 1 disabled menu item', () => {
    const element = mount(<NavbarLink {...propsDisabledLink} />);
    expect(element.find('.nav-item .disabled').length).toBe(1);
  });
});
