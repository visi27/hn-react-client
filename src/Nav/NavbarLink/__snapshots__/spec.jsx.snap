// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav has valid snapshot 1`] = `
<li
  className="nav-item active"
>
  <a
    className="nav-link "
    href="/"
    onClick={[Function]}
  >
    testLink
  </a>
</li>
`;
